app/models/concerns/filterable.rb

```diff
@@ -0,0 +1,13 @@
+module Filterable
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    def filter(filtering_params)
+      results = self.where(nil)
```

Comment:
This method would be simplify the searching method and this method chain together the scopes in the the User model http://www.justinweiss.com/blog/2014/02/17/search-and-filter-rails-models-without-bloating-your-controller/ Yes it is chainable as you can see in the user controller https://github.com/AgileVentures/WebsiteOne/pull/617/files#diff-4e05ad0d64e6100656b63ad1e78f32c5R8